src/apps/wallet/sign_tx/signing.py

```diff
@@ -275,8 +275,9 @@ class SigningError(ValueError):
                             multisig_get_pubkeys(txi_sign.multisig),
                             txi_sign.multisig.m)
                     elif txi_sign.script_type == InputScriptType.SPENDADDRESS:
-                        txi_sign.script_sig = output_script_p2pkh(
-                            ecdsa_hash_pubkey(key_sign_pub))
+                        txi_sign.script_sig = output_script_p2pkh(ecdsa_hash_pubkey(key_sign_pub))
+                        if(coin.bip115 and txi_sign.prev_block_hash_bip115 is not None):
+                            txi_sign.script_sig += script_replay_protection_bip115(txi_sign.prev_block_hash_bip115, txi_sign.prev_block_height_bip115)
```

Comment:
the fields `prev_block_hash_bip115` and `prev_block_height_bip115` are extracted from previous output's `scriptPubKey` apparently -- basically reversing what `script_replay_protection_bip115` does.

esp. given that this coin is not segwit-based and the prev output was actually requested somewhere near here (in `check_tx_fee`), ISTM the `prev_*_bip115` fields should be removed and Trezor should instead extract them itself